extension/source/background/TransactionsController.ts

```diff
@@ -150,44 +146,44 @@ export default class TransactionsController {
       const { outgoing: transactions } = await this.transactions.read();
       const transaction = transactions.find((t) => t.id === id);
 
-      if (transaction) {
-        switch (status) {
-          // move to APPROVED or REJECTED only if status was NEW
-          case TransactionStatus.APPROVED:
-            if (transaction?.status === TransactionStatus.NEW) {
-              transaction.status = status;
-            }
-            break;
-          case TransactionStatus.REJECTED:
-            if (transaction?.status === TransactionStatus.NEW) {
-              transaction.status = status;
-            }
-            break;
-          // move to CANCELLED only if status was APPROVED
-          case TransactionStatus.CANCELLED:
-            if (transaction?.status === TransactionStatus.APPROVED) {
-              transaction.status = status;
-            }
-            break;
-          // move to CONFIRM or FAILED only if status was APPROVED
-          case TransactionStatus.CONFIRMED:
-            if (transaction?.status === TransactionStatus.APPROVED) {
-              transaction.status = status;
-            }
-            break;
-          case TransactionStatus.FAILED:
-            if (transaction?.status === TransactionStatus.APPROVED) {
-              transaction.status = status;
-            }
-            break;
-          default:
-            break;
-        }
-
-        const updatedTx = transactions.filter((t) => t.id !== id);
-        updatedTx.push(transaction);
-        await this.transactions.update({ outgoing: updatedTx });
-      }
+      assert(
+        transaction !== undefined,
+        () => new Error('Transaction not found'),
+      );
+
+      const validTransitions: Partial<
+        Record<TransactionStatus, TransactionStatus[]>
+      > = {
+        [TransactionStatus.NEW]: [
+          TransactionStatus.APPROVED,
+          TransactionStatus.REJECTED,
+        ],
+        [TransactionStatus.APPROVED]: [
+          TransactionStatus.CANCELLED,
+          TransactionStatus.APPROVED,
+          TransactionStatus.FAILED,
+        ],
+      };
+
+      const currentValidTransitions =
+        validTransitions[transaction.status] ?? [];
+
+      assert(
+        currentValidTransitions.includes(status),
+        () =>
+          new Error(
+            [
+              `Forbidden transaction status transition:`,
+              `${transaction.status} -> ${status}`,
+            ].join(' '),
+          ),
```

Comment:
I believe it's important to assert that the transition is valid rather than simply ignoring invalid transitions as before. Is there a reason to prefer ignoring invalid transitions?